docs/addons/istio-cni.md

```diff
@@ -0,0 +1,61 @@
+# Istio Cni Add-on
+
+Istio is an open platform for providing a uniform way to integrate microservices, manage traffic flow across microservices, enforce policies and aggregate telemetry data. The Istio CNI plugin performs the Istio mesh pod traffic redirection in the Kubernetes pod lifecycle’s network setup phase, thereby removing the requirement for the NET_ADMIN and NET_RAW capabilities for users deploying pods into the Istio mesh. 
+
+***IMPORTANT***:
+
+1. This add-on depends on [Istio Base](istio-base.md) and [istio Control Plane](istio-control-plane.md) Add-ons for cluster-wide resources and CRDs.
+
+***Istio Base add-on and Istio Control Plane addon-on must be present in add-on array*** and ***must be in add-on array before the Istio Cni add-on*** for it to work, as shown in below example. Otherwise will run into error `Assertion failed: Missing a dependency for IstioBaseAddOn`.
+
+## Usage
+
+Add the following as an add-on to your main.ts file to add Istio Control Plane to your cluster
```

Comment:
let's figure out what default settings are applied based on the helm chart. I assume that this addon is not replacing VPC-CNI - if so we need to call it out. 
I see that istio cni addon has the setting values.cni.chained. I assume it must be true for it to work on AWS. Let's call it out and make it configurable.

Reply:
First part is done, unclear on the second part, lets discuss on this design!

Reply:
Done.